sphinx/tutorial/cylc/furthertopics/clock-triggered-tasks.rst

```diff
@@ -0,0 +1,179 @@
+the first of January 2000 we would write ``2000-01-01T00Z``).
+
+We now have a simple suite with a single task that prints "bong" a number
+of times equal to the (cycle point) hour.
+
+Run your suite using::
+
+   cylc run clock-trigger
+
+Stop the suite after a few cycles using the :guilabel:`stop` button in the
+``cylc gui``. Notice how the tasks run as soon as possible rather than
+waiting for the actual time to be equal to the cycle point.
+
+
+Clock Triggering Tasks
+----------------------
+
+We want our clock to only ring in real time rather than the simulated
+cycle time.
+
+To do this, add the following lines to the ``[scheduling]`` section of
+your ``suite.rc``:
+
+.. code-block:: cylc
+
+   [[special tasks]]
+       clock-trigger = bell(PT0M)
+
+This tells the suite to clock trigger the ``bell`` task with a cycle
+offset of ``0`` hours.
+
+Save your changes and run your suite::
+
+   cylc run clock-trigger
+
+
+Results
+-------
+
+Your suite should now be running the bell task in realtime. Any cycle times
+that have already passed (such as the one defined by ``initial cycle time``)
+will be run as soon as possible, while those in the future will wait for that
+time to pass.
+
+At this point you may want to leave your suite running until the next hour
+has passed in order to confirm the clock triggering is working correctly.
+Once you are satisfied, stop your suite.
+
+By making the ``bell`` task a clock triggered task we have made it run in
+realtime. Thus, when the time caught up with the cycle time, the bell task
+triggered.
+
+
+Further Clock Triggering
+------------------------
+
+We will now modify our suite to run tasks at quarter-past, half-past and
+quarter-to the hour.
+
+Open your ``suite.rc`` and modify the ``[runtime]`` section by adding the
+following:
+
+.. code-block:: cylc
+
+   [[quarter_past, half_past, quarter_to]]
+       script = echo 'chimes'
+
+Similarly, modify the ``[[scheduling]]`` section as follows:
+
+.. code-block:: cylc
+
+   [[special tasks]]
+       clock-trigger = bell(PT0M), quarter_past(PT15M), half_past(PT30M), quarter_to(PT45M)
+   [[dependencies]]
+       [[[PT1H]]]
+           graph = bell => quarter_past => half_past => quarter_to
+
+Note the different values used for the cycle offsets of the clock-trigger tasks.
+
+Save your changes and run your suite using::
+
+   cylc run clock-trigger now
+
+.. note::
+
+   The ``now`` argument will run your suite using the current time for the
+   initial cycle point.
+
+Again, notice how the tasks trigger until the current time is reached.
+
+Leave your suite running for a while to confirm it is working as expected
+and then shut it down using the :guilabel:`stop` button in the ``cylc gui``.
+
+
+Summary
+-------
+
+You have now successfully created and run a suite that:
+
+* Runs a bell task in realtime on the hour
+* Runs different chiming tasks at quarter-past, half-past and quarter-to the
+  hour
+
+For more information see the `cylc User Guide`_.
```

Comment:
Might be worth mentioning and/or linking the family and suicide trigger pages to highlight the other possibilities for advanced/non-standard triggers.

Reply:
Would love to. Unfortunately the cylc documentation is not properly linkable. One day we will probably change the documentation system and will be able to update this.